streamingpro-mlsql/src/main/java/tech/mlsql/ets/Ray.scala

```diff
@@ -104,20 +104,21 @@ class Ray(override val uid: String) extends SQLAlg with VersionCompatibility wit
 
     val timezoneID = session.sessionState.conf.sessionLocalTimeZone
     val df = session.table(sourceTable)
-
+    val maxIterCount = runnerConf.getOrElse("maxIterCount", "1000").toInt
+    session.conf.set("maxIterCount", maxIterCount.toString)
     // start spark data servers for model if user configure model table in et params.
     val modelTableOpt = etParams.get("model")
     if (modelTableOpt.isDefined) {
       val modelDf = session.table(modelTableOpt.get)
       val modelServer = new MasterSlaveInSpark("temp-model-server-in-spark", session, context.owner)
-      modelServer.build(modelDf, MasterSlaveInSpark.defaultDataServerImpl)
+      modelServer.build(modelDf, MasterSlaveInSpark.defaultDataServerWithIterCountImpl)
```

Comment:
Use session to store the maxIterCount  is not a good idea. Try to modify the signature of `MasterSlaveInSpark.build` , you can add a new parameter like this：

```scala
modelServer.build(modelDf, MasterSlaveInSpark.defaultDataServerWithIterCountImpl,runnerConf)
```

Reply:
DONE